django_mongodb/expressions.py

```diff
@@ -1,4 +1,33 @@
-from django.db.models.expressions import Col, ExpressionWrapper, Value
+import datetime
+from decimal import Decimal
+
+from bson import Decimal128
+from django.core.exceptions import EmptyResultSet, FullResultSet
+from django.db.models.expressions import Case, Col, ExpressionWrapper, Value, When
+
+
+def case(self, compiler, connection):
+    case_parts = []
+    default_mql = self.default.as_mql(compiler, connection)
```

Comment:
I think this one could be in the else part of the for

Reply:
I think you're right. I followed the pattern in `Case.as_sql()` so let's see if this change is accepted upstream: https://github.com/django/django/pull/18296.